syncer/job.go

```diff
@@ -109,27 +109,22 @@ func newDMLJob(tp opType, sourceSchema, sourceTable, targetSchema, targetTable,
 // newDDL job is used to create a new ddl job
 // when cfg.ShardMode == "", ddlInfo == nil，sourceTbls != nil, we use sourceTbls to record ddl affected tables.
 // when cfg.ShardMode == ShardOptimistic || ShardPessimistic, ddlInfo != nil, sourceTbls == nil.
-func newDDLJob(ddlInfo *shardingDDLInfo, ddls []string, location, startLocation, cmdLocation binlog.Location,
-	sourceTbls map[string]map[string]struct{}, originSQL string, eventHeader *replication.EventHeader) *job {
+func newDDLJob(qec *queryEventContext) *job {
 	j := &job{
 		tp:        ddl,
-		ddls:      ddls,
-		originSQL: originSQL,
+		ddls:      qec.needHandleDDLs,
+		originSQL: qec.originSQL,
 
-		location:        location,
-		startLocation:   startLocation,
-		currentLocation: cmdLocation,
-		eventHeader:     eventHeader,
+		location:        *qec.lastLocation,
+		startLocation:   *qec.startLocation,
+		currentLocation: *qec.currentLocation,
+		eventHeader:     qec.header,
 		jobAddTime:      time.Now(),
 	}
 
-	if ddlInfo != nil {
-		j.sourceTbl = map[string][]string{ddlInfo.tables[0][0].Schema: {ddlInfo.tables[0][0].Name}}
-		j.targetSchema = ddlInfo.tables[1][0].Schema
-		j.targetTable = ddlInfo.tables[1][0].Name
-	} else if sourceTbls != nil {
-		sourceTbl := make(map[string][]string, len(sourceTbls))
-		for schema, tbMap := range sourceTbls {
+	if qec.sourceTbls != nil && len(qec.sourceTbls) != 0 {
```

Comment:
seems dont't need this check? `&& len(qec.sourceTbls) != 0` , the for loop in L127 is already do this